contracts/contracts/Poll.sol

```diff
@@ -159,7 +155,8 @@ contract Poll is Params, Utilities, SnarkCommon, IPoll {
   /// @inheritdoc IPoll
   function publishMessage(Message memory _message, PubKey calldata _encPubKey) public virtual isWithinVotingDeadline {
     // we check that we do not exceed the max number of messages
-    if (numMessages >= maxValues.maxMessages) revert TooManyMessages();
+    uint256 maxMessages = TREE_ARITY ** treeDepths.messageTreeDepth;
```

Comment:
rather than calculating all the times this can be an immutable variable and calculated in the constructor